yt/frontends/boxlib/data_structures.py

```diff
@@ -1354,24 +1284,30 @@ def _parse_parameter_file(self):
         self.cosmological_simulation = 1
 
         jobinfo_filename = os.path.join(self.output_dir, "job_info")
-        line = ""
+
+        has_cosmo_info = False
         with open(jobinfo_filename, "r") as f:
-            while not line.startswith(" Cosmology Information"):
+            for line in f:
                 # get the code git hashes
                 if "git hash" in line:
                     # line format: codename git hash:  the-hash
                     fields = line.split(":")
                     self.parameters[fields[0]] = fields[1].strip()
-                line = next(f)
 
-            # get the cosmology
-            for line in f:
-                if "Omega_m (comoving)" in line:
-                    self.omega_matter = float(line.split(":")[1])
-                elif "Omega_lambda (comoving)" in line:
-                    self.omega_lambda = float(line.split(":")[1])
-                elif "h (comoving)" in line:
-                    self.hubble_constant = float(line.split(":")[1])
+                if line.startswith(" Cosmology Information"):
+                    has_cosmo_info = True
+                    break
+
+            if has_cosmo_info:
+                for line in f:
+                    if "Omega_m (comoving)" in line:
+                        self.omega_matter = float(line.split(":")[1])
+                    elif "Omega_lambda (comoving)" in line:
+                        self.omega_lambda = float(line.split(":")[1])
+                    elif "h (comoving)" in line:
+                        self.hubble_constant = float(line.split(":")[1])
+            else:
+                raise NotImplementedError
```

Comment:
@zingale If nothing else is broken, I would still like to finish the PR with default parameters here. The dataset we have that hits this  is `nyx_small` :)

Reply:
sounds good.

I think in the future, I'll do a PR to add some additional MAESTROeX stuff to the frontend, since the format has changed slightly.  But at least with this change detecting it as a MAESTROeX dataset, yt won't get the velocity stuff wrong.

Reply:
so do you know what would be the defaults here ? or if it is in fact possible that this Nyx simulation is not cosmological ?

Reply:
I'm really not a Nyx person, but @jmsexton03 is.  Maybe she can answer this?

Reply:
I don't know what `nyx_small` is, I do know that the job_info files from Nyx plotfiles I generate will have those cosmology fields in them. I'm not clear on what's broken here.

Reply:
looking at that file, it looks like it was produced in `Nyx/Exec/simple_output`, but that directory no longer exists in Nyx.  The file was originally output in 2012, so a lot must have changed.

Reply:
Nyx now also includes an omega_radiation cosmology term (almost always 0). If we add that number to job_info in a similar way, can yt store it in the same data structure?
https://github.com/yt-project/yt/blob/fdc9210a35e41b45c114f271798ea46c2179d2eb/yt/utilities/cosmology.py#L41

Reply:
Sounds like yt is very unlikely to ever come across these outdated format then. I can probably just make this a special case with `cosmological_simulation = 0`

Reply:
indeed, I suspect a lot of the sample data is quite outdated.  Maybe we should have a push (separate from this PR) to update some of the datasets to current code output.

Reply:
Nyx simulations have had cosmology fields in job_info since 2016 when Nyx was first on github, so it's highly unlikely a Nyx plotfile without cosmology fields in job_info will be used now. It looks like your fix should handle it well.